scripts/algorithms/common/buffer/replay_buffer.py

```diff
@@ -84,3 +87,88 @@ def sample(
     def __len__(self) -> int:
         """Return the current size of internal memory."""
         return len(self.buffer)
+
+
+class NStepTransitionBuffer:
+    """Fixed-size buffer to store experience tuples.
+
+    Attributes:
+        buffer (list): list of replay buffer
+        buffer_size (int): buffer size not storing demos
+        demo_size (int): size of a demo to permanently store in the buffer
+        cursor (int): position to store next transition coming in
+
+    """
+
+    def __init__(self, buffer_size: int, n_step: int, gamma: float, demo: list = None):
+        """Initialize a ReplayBuffer object.
+
+        Args:
+            buffer_size (int): size of replay buffer for experience
+            demo (list): demonstration transitions
+
+        """
+        assert buffer_size > 0
+
+        self.n_step_buffer: Deque = deque(maxlen=n_step)
+        self.buffer_size = buffer_size
+        self.buffer: list = list()
+        self.n_step = n_step
+        self.gamma = gamma
+        self.demo_size = 0
+        self.cursor = 0
+
+        # if demo exists
+        if demo:
+            self.demo_size = len(demo)
+            self.buffer.extend(demo)
+
+        self.buffer.extend([None] * self.buffer_size)
+
+    def add(self, transition: Tuple[np.ndarray, ...]) -> Tuple[Any, ...]:
+        """Add a new transition to memory."""
+        self.n_step_buffer.append(transition)
+
+        # single step transition is not ready
+        if len(self.n_step_buffer) < self.n_step:
+            return ()
+
+        # add a multi step transition
+        reward, next_state, done = get_n_step_info(self.n_step_buffer, self.gamma)
+        curr_state, action = transition[:2]
```

Comment:
Sorry, should modify too :(
`curr_state, action = self.n_step_buffer[0][:2]`

Reply:
Fixed it in commit 3263cc5. Thank you!